UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -246,6 +236,32 @@ public AgentInfo Info
 
         WriteAdapter m_WriteAdapter = new WriteAdapter();
 
+
+        RewardProviderComponent m_RewardProviderComponent;
```

Comment:
I don't have strong feelings either way, but if we want to be more like what we do for sensors, we should just store the IRewardProvider here (and in `InitializeRewardProvider()`)